docs/guide/api-environment-plugins.md

```diff
@@ -151,6 +152,35 @@ const UnoCssPlugin = () => {
 }
 ```
 
+If a plugin isn't environment aware and has state that isn't keyed on the current environment (as it is common in Rollup build plugins like `rollup-plugin-visualizer`), the `applyToEnvironment` hook allows to easily make it per-environment.
```

Comment:
Trying to understand this example better, `rollup-plugin-visualizer` only uses the [`generateBundle` hook](https://github.com/btd/rollup-plugin-visualizer/blob/26d732e0742c7a99cacb7c74a61682806ce6989e/plugin/index.ts#L135) and doesn't use any shared cache. Do we need to wrap it with `applyToEnvironment`?

Unless I'm misunderstanding something, rollup plugins that base solely on `renderChunk`, `generateBundle`, etc are already bound to a specific Rollup output, so any data should be cached using the output as the key. And since each environment has their own outputs, the Rollup plugin should already indirectly be isolated per environment?

I haven't tested `rollup-plugin-visualizer` locally so I might be wrong. (I also wonder if `this.cache` helps here)